src/main/java/emu/grasscutter/server/dispatch/DispatchServer.java

```diff
@@ -177,23 +177,23 @@ public void start() throws Exception {
 			SSLContext sslContext = SSLContext.getInstance("TLS");
 			try (FileInputStream fis = new FileInputStream(Grasscutter.getConfig().getDispatchOptions().KeystorePath)) {
 				char[] keystorePassword = Grasscutter.getConfig().getDispatchOptions().KeystorePassword.toCharArray();
-
+				KeyManagerFactory _kmf;
 				try {
 					KeyStore ks = KeyStore.getInstance("PKCS12");
 					ks.load(fis, keystorePassword);
 					KeyManagerFactory kmf = KeyManagerFactory.getInstance("SunX509");
+					_kmf = kmf;
 					kmf.init(ks, keystorePassword);
 				} catch (Exception e) {
 					Grasscutter.getLogger().warn("[Dispatch] Unable to load keystore. Using default keystore password...");
 					KeyStore ks = KeyStore.getInstance("PKCS12");
 					ks.load(fis, "123456".toCharArray());
 					KeyManagerFactory kmf = KeyManagerFactory.getInstance("SunX509");
 					kmf.init(ks, "123456".toCharArray());
-				} catch (Exception e) {
-					Grasscutter.getLogger().warn("[Dispatch] Error while loading keystore!");
```

Comment:
I think we still need this.

It's a bit ugly, but maybe a:

```java
catch (Exception originalEx) {
    try {
        // try to initialize kmf with "123456"
    }
    catch (Exception _) {
        Grasscutter.getLogger().warn("[Dispatch] Error while loading keystore!");
        originalEx.printStackTrace();
        throw originalEx; // don't care about the exception for the "123456" default password attempt
    }
}
```

Reply:
That will result in an error because Exception e was already catched before, and I don't know all of the error classes